### PR TITLE
Squeeze op

### PR DESCRIPTION
**Pull request by reminisce**
## Description ##
This PR implemented squeeze op in MXNet as numpy.squeeze. Requested by @szha .
https://docs.scipy.org/doc/numpy/reference/generated/numpy.squeeze.html

@eric-haibin-lin 

### Essentials ###
- [x] Passed code style checking (`make lint`)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

**Comment by chinakook** [comment]
How about unsqeeze?

**Comment by reminisce** [comment]
@chinakook `expand_dims`?

File: src/operator/tensor/matrix_op.cc [hunk 1]
```diff
@@ -739,5 +740,43 @@ NNVM_REGISTER_OP(_backward_stack)
...
+.set_num_inputs(1)
+.set_num_outputs(1)
+.set_attr_parser(ParamParser<SqueezeParam>)
+.set_attr<nnvm::FListInputNames>("FListInputNames",
+  [](const NodeAttrs& attrs) {
+    return std::vector<std::string>{"data"};
+  })
+.set_attr<nnvm::FInferShape>("FInferShape", SqueezeShape)
+.set_attr<nnvm::FInferType>("FInferType", ElemwiseType<1, 1>)
+.set_attr<FCompute>("FCompute<cpu>", SqueezeCompute<cpu>)
```
Comment (piiswrong) [inline src/operator/tensor/matrix_op.cc]: I think you can use IdentityCompute directly?

Comment (reminisce) [reply src/operator/tensor/matrix_op.cc]: You are right. Changed now. Thanks.

**Comment by szha** [comment]
@reminisce could you add a fluent entry for this in both nd and sym?

**Comment by reminisce** [comment]
@szha Oh, yes. I forgot to do so. I will submit a PR to add the fluent entry for this op.

**Comment by chinakook** [comment]
@reminisce  yes, It's equalevant.

**Pull request merged**